python/ray/ml/examples/lightgbm_example.py

```diff
@@ -0,0 +1,110 @@
+import argparse
+from typing import Tuple
+
+import pandas as pd
+
+import ray
+from ray.ml.checkpoint import Checkpoint
+from ray.ml.predictors.integrations.lightgbm import LightGBMPredictor
+from ray.ml.train.integrations.lightgbm import LightGBMTrainer
+from ray.data.dataset import Dataset
+from ray.ml.result import Result
+from ray.ml.constants import TRAIN_DATASET_KEY
+from ray.ml.preprocessors import StandardScaler
+from sklearn.datasets import load_breast_cancer
+from sklearn.model_selection import train_test_split
+
+
+def prepare_data() -> Tuple[Dataset, Dataset, Dataset]:
+    data_raw = load_breast_cancer()
+    dataset_df = pd.DataFrame(data_raw["data"], columns=data_raw["feature_names"])
+    dataset_df["target"] = data_raw["target"]
+    train_df, test_df = train_test_split(dataset_df, test_size=0.3)
+    train_dataset = ray.data.from_pandas(train_df)
+    valid_dataset = ray.data.from_pandas(test_df)
+    test_dataset = ray.data.from_pandas(test_df.drop("target", axis=1))
+    return train_dataset, valid_dataset, test_dataset
+
+
+def train_lightgbm(num_workers: int, use_gpu: bool = False) -> Result:
+    train_dataset, valid_dataset, _ = prepare_data()
+
+    # Scale some random columns
+    columns_to_scale = ["mean radius", "mean texture"]
+    preprocessor = StandardScaler(columns=columns_to_scale)
+
+    # LightGBM specific params
+    params = {
+        "objective": "binary",
+        "metric": ["binary_logloss", "binary_error"],
+    }
+
+    trainer = LightGBMTrainer(
+        scaling_config={
+            "num_workers": num_workers,
+            "use_gpu": use_gpu,
+        },
+        label_column="target",
+        params=params,
+        datasets={TRAIN_DATASET_KEY: train_dataset, "valid": valid_dataset},
```

Comment:
I understand the logic of defining global variables for safety and to not have too many magic keys flying around here, but given that we tend to expose these examples to users in docs etc., I'd rather go for readability here and use this technique in tests only

Reply:
i.e. just use "train" here